docs/permission/plugin-authors/authorize-the-update-endpoint.md

```diff
@@ -0,0 +1,146 @@
++     if (decision.result === AuthorizeResult.ALLOW) {
++       res.json(update(req.body));
++       return;
++     }
++     throw new NotAllowedError('Unauthorized');
+    });
+```
+
+**Important:** Notice that we are passing an extra `resourceRef` object, containing the `id` of the todo we want to authorize. This enables decisions based on characteristics of the resource, but it's important to note that to enable authorizing multiple resources at once, **the resourceRef is not passed to PermissionPolicy#handle**. Instead, policies must return a _conditional decision_.
+
+Before diving into the extra steps needed for supporting conditional decisions, let's go back to the permission policy's handle function used by your adopters and try to authorize our new permission.
+
+Let's edit `packages/backend/src/plugins/permission.ts`
+
+```diff
+    if (request.permission.name === 'todos.list.create') {
+      return {
+        result: AuthorizeResult.DENY,
+      };
+    }
++   if (request.permission.resourceType === 'todo-item') {
++     if (request.permission.attributes.action === 'update') {
++       return {
++         result: AuthorizeResult.CONDITIONAL,
++         pluginId: 'todolist',
++         resourceType: 'todo-item',
++         conditions: {
++           rule: 'IS_OWNER',
++           params: [user?.identity.userEntityRef],
++         },
++       };
++     }
++   }
+    return {
+      result: AuthorizeResult.ALLOW,
+    };
+```
+
+This is what happens when a _Conditional Decision_ is returned. We are saying:
+
+> Hey permission framework, I can't make a decision alone. Please go to the plugin with id `todolist` and ask it to apply these conditions.
+
+Now if we try to edit an item from the UI, we should spot the following error in the backend's console:
+
+```
+backstage error Unexpected response from plugin upstream when applying conditions.
+Expected 200 but got 404 - Not Found type=errorHandler stack=Error:
+Unexpected response from plugin upstream when applying conditions. Expected 200 but got 404 - Not Found
+```
+
+This happens because our plugin should have exposed a specific endpoint, used by the permission framework to apply conditional decisions. The new endpoint should also be able to support some conditions. In our case, `IS_OWNER` is the only type of condition we want to support.
+
+Create a new `plugins/todo-list-backend/src/service/rules.ts` file and append the following code:
+
+```diff
++ import { makeCreatePermissionRule } from '@backstage/plugin-permission-node';
++ import { Todo } from './todos';
+
++ const createTodoListPermissionRule = makeCreatePermissionRule<
++   Todo,
++   undefined
++ >();
+
++ export const isOwner = createTodoListPermissionRule({
++   name: 'IS_OWNER',
++   description: 'Should allow only if the todo belongs to the user',
++   apply: (resource, userId) => {
++     return resource.author === userId;
++   },
++   toQuery: userId => {
++     throw new Error('toQuery not implemented');
```

Comment:
The reason why I chose to provide a partial implementation here is based on my experience when I was learning the permission framework. It was hard for me to understand when each one of the functions was executed.

By providing an initial partial implementation, I could focus on the exact update todo and later on fully implement the feature when authorizing the read todo.

Reply:
I like the "one thing at a time" approach 👍  let's just make sure to include a comment saying that we'll come back to this later as readers may wonder.

Reply:
yes, good point. Let me fix this